plugins/english/lnmtl.ts

```diff
@@ -0,0 +1,224 @@
+import { load as parseHTML } from "cheerio";
+import { fetchApi, fetchFile } from "@libs/fetch";
+import { FilterTypes, Filters } from "@libs/filterInputs";
+import { Plugin } from "@typings/plugin";
+
+class LnMTLPlugin implements Plugin.PluginBase {
+    id = "lnmtl";
+    name = "LnMTL";
+    icon = "src/en/lnmtl/icon.png";
+    site = "https://lnmtl.com/";
+    version = "1.0.0";
+
+    async popularNovels(
+        page: number,
+        { filters }: Plugin.PopularNovelsOptions<typeof this.filters>
+    ): Promise<Plugin.NovelItem[]> {    
+        let link = this.site + "novel?";
+
+        link += `orderBy=${filters.order.value}`;
+        link += `&order=${filters.sort.value}`;
+        link += `&filter=${filters.storyStatus.value}`;
+        link += `&page=${page}`;                         
+
+        const headers = new Headers();
+        const body = await fetchApi(link, { headers }).then((result) =>
+            result.text()
+        );
+
+        const loadedCheerio = parseHTML(body);
+        const novels: Plugin.NovelItem[] = [];
+
+        loadedCheerio(".media").each(function () {
+            const novelName = loadedCheerio(this).find("h4").text();
+            const novelCover = loadedCheerio(this).find("img").attr("src");
+            const novelUrl = loadedCheerio(this).find("h4 > a").attr("href");
+
+            if (!novelUrl) return;
+
+            const novel = {
+                name: novelName,
+                cover: novelCover,
+                url: novelUrl,
+            };
+
+            novels.push(novel);
+        });
+
+        return novels;
+    };
+
+    async parseNovelAndChapters(novelUrl: string): Promise<Plugin.SourceNovel> {
+        const url = novelUrl;
+        const headers = new Headers();
+        const result = await fetchApi(url, { headers });
+        const body = await result.text();
+
+        const loadedCheerio = parseHTML(body);
+
+        const novel: Plugin.SourceNovel = {
+            url,
+            chapters: [],
+        };
+
+        novel.name = loadedCheerio(".novel-name").text();
+
+        novel.cover = loadedCheerio("div.novel").find("img").attr("src");
+
+        novel.summary = loadedCheerio("div.description").text().trim();
+
+        loadedCheerio(".panel-body > dl").each(function () {
+            const detailName = loadedCheerio(this).find("dt").text().trim();
+            const detail = loadedCheerio(this).find("dd").text().trim();
+
+            switch (detailName) {
+                case "Authors":
+                    novel.author = detail;
+                    break;
+                case "Current status":
+                    novel.status = detail;
+                    break;
+            }
+        });
+
+        novel.genres = loadedCheerio('.panel-heading:contains(" Genres ")')
+            .next()
+            .text()
+            .trim()
+            .replace(/\s\s/g, ",");
```

Comment:
are you sure this correct?
what there's novel with genres: `Yuri` `Comedy` `Adapted to Anime`

Reply:
output from this command for Cultivation Chat Group https://lnmtl.com/novel/cultivation-chat-group
``"genres": "Action,Adventure,Comedy,Fantasy,Martial Arts,Mystery,School Life,Slice of Life,Xianxia"``

Reply:
updated code